src/SFA.DAS.LevyTransferMatching/Application/Commands/CreditPledge/CreditPledgeCommandHandler.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.LevyTransferMatching.InnerApi.LevyTransferMatching.Requests;
+using SFA.DAS.LevyTransferMatching.Interfaces;
+
+namespace SFA.DAS.LevyTransferMatching.Application.Commands.CreditPledge
+{
+    public class CreditPledgeCommandHandler : IRequestHandler<CreditPledgeCommand, CreditPledgeCommandResult>
+    {
+        private readonly ILevyTransferMatchingService _levyTransferMatchingService;
+        private readonly ILogger<CreditPledgeCommandHandler> _logger;
+
+        public CreditPledgeCommandHandler(ILevyTransferMatchingService levyTransferMatchingService, ILogger<CreditPledgeCommandHandler> logger)
+        {
+            _levyTransferMatchingService = levyTransferMatchingService;
+            _logger = logger;
+        }
+
+        public async Task<CreditPledgeCommandResult> Handle(CreditPledgeCommand request, CancellationToken cancellationToken)
+        {
+            if (request.Amount <= 0)
+            {
+                _logger.LogInformation($"Amount is 0 for pledge {request.PledgeId} - pledge will *not* be credited");
+
+                return null;
```

Comment:
See other comment

Reply:
I've included a property that indicates whether the `CreditPledge` endpoint was called.